DopplerCustomDomain.Test/CustomDomainEndpointTest.cs

```diff
@@ -393,5 +418,113 @@ public async Task GET_domain_ip_resolution_should_return_BadRequest_when_it_reso
+
+            using var appFactory = _factory.WithBypassAuthorization();
+
+            var client = CreateHttpClient(
+                appFactory,
+                customDomainProviderService: customDomainProviderServiceMock.Object,
+                dnsResolutionValidator: dnsResolutionValidatorMock.Object);
+
+            var request = new HttpRequestMessage(HttpMethod.Put, $"http://localhost/{domainName}");
+            request.Content = JsonContent.Create(new { service, ruleType });
+
+            // Act
+            var response = await client.SendAsync(request);
+
+            // Assert
+            Assert.NotNull(response);
+            Assert.Equal(HttpStatusCode.BadRequest, response.StatusCode);
+            customDomainProviderServiceMock.Verify(x => x.CreateCustomDomain(It.IsAny<string>(), It.IsAny<string>(), It.IsAny<RuleType>()), Times.Never);
+        }
+
+        [Fact]
+        public async Task PUT_domain_should_store_domain_when_it_does_not_resolve_to_our_IP_but_validation_is_ignored()
+        {
+            // Arrange
+            var fixture = new Fixture();
+            var domainName = fixture.Create<string>();
+            var expectedRuleType = RuleType.HttpsOnly;
+            var ruleType = expectedRuleType.ToString();
+            var expectedService = "relay-actions-api_service_prod@docker";
+            var service = "relay-tracking";
+            var ipResolutionValidation = "ignored";
+
+            var customDomainProviderServiceMock = CreateCustomDomainProviderServiceMock();
+            var dnsResolutionValidatorMock = CreateDnsResolutionValidatorMock();
+            dnsResolutionValidatorMock.Setup(x => x.IsNamePointingToOurServiceAsync(domainName)).ReturnsAsync(false);
+
+            var customDomainControllerLoggerMock = new Mock<ILogger<CustomDomainController>>();
+
+            using var appFactory = _factory.WithBypassAuthorization();
+
+            var client = CreateHttpClient(
+                appFactory,
+                customDomainProviderService: customDomainProviderServiceMock.Object,
+                dnsResolutionValidator: dnsResolutionValidatorMock.Object,
+                customDomainControllerLogger: customDomainControllerLoggerMock.Object);
+
+            var request = new HttpRequestMessage(HttpMethod.Put, $"http://localhost/{domainName}");
+            request.Content = JsonContent.Create(new { service, ruleType, ipResolutionValidation });
+
+            // Act
+            var response = await client.SendAsync(request);
+
+            // Assert
+            Assert.NotNull(response);
+            Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+            customDomainProviderServiceMock.Verify(x => x.CreateCustomDomain(domainName, expectedService, expectedRuleType), Times.Once);
+            customDomainProviderServiceMock.Verify(x => x.CreateCustomDomain(It.IsAny<string>(), It.IsAny<string>(), It.IsAny<RuleType>()), Times.Once);
```

Comment:
This could be simplify to only one assert
```suggestion
            customDomainProviderServiceMock.Verify(x => x.CreateCustomDomain(It.Is<string>(x => x == domainName), It.Is<string>(x => x == expectedService), It.Is<RuleType>(x => x == expectedRuleType)), Times.Once);
```

Reply:
Your suggestion works in the same way as my first line.

The second one is to verify that we are not saving anything else.

Reply:
In that case, maybe should be another test. 🤔

Reply:
Me parece que estás tomando demasiado literal esa recomendación.